### PR TITLE
Support large integer stringification thresholds in both encoders

### LINKED ISSUE #306: Support 128-bit integers using int_as_string_bitcount
Currently `int_as_string_bitcount` supports only up to 63bits.  Please add support for any bitlength or at least 127(128) bits.  [Snowflake](https://docs.snowflake.com/en/sql-reference/data-types-numeric.html#int-integer-bigint-smallint-tinyint-byteint) supports 128bit integers, currently any bigints that I load have to be converted to a string if they are less than 64bits. 

ideally all of the following should be stored as a number given the following code:
```python 
test = {
    "max_int": 99999999999999999999999999999999999999,
    "63 bits": 999999999999999999,
    "53 bits": 999999999999999,
    "overflow": 999999999999999999999999999999999999991,
}
result = simplejson.dumps(test, int_as_string_bitcount=128)

print(result)
```

### PR DESCRIPTION
**Pull request by be-student**
Allow `int_as_string_bitcount` thresholds of 64 bits and above in the C encoder, including 128-bit integers. Keep the small-threshold comparison path, and use magnitude bit lengths for larger thresholds so even very large counts do not allocate a power-of-two boundary. The Python encoder uses the same magnitude rule.

Closes #306. Tests cover signed boundaries through 1,024 bits, nested values, indented output, and counts larger than native integers. Documentation and the encoder ownership notes are updated.

Validation: the original C encoder rejects the new boundary cases. The full C/Python runner passes on Python 3.12, 3.14 release and 3.14 debug (462 tests each), with explicit C-extension wiring checks. Strict C89 and warnings-as-errors builds pass on 3.12, 3.14 release and 3.14 debug. Legacy Python 2.7, free-threaded builds and the remaining platform matrix are left to CI.

### CODE REVIEW

**Review by etrepum — COMMENTED** [review]
Codex found an edge case in this one:

Reviewed e709f3b. Found one edge-case correctness issue.

[P2] Normalize integer subclasses once before formatting and checking their magnitude.
At _speedups.c:716–721, PyNumber_Long(obj) invokes __int__ again after encoder_long_to_str() has already converted the object. A stateful conversion can therefore make the quoting decision use a different integer from the one being serialized:

```python
class ChangingInt(int):
    calls = 0

    def __int__(self):
        self.calls += 1
        return (1 << 64) if self.calls == 1 else 1

simplejson.dumps(ChangingInt(1), int_as_string_bitcount=64)
```

Confirmed results:

PR C encoder: 18446744073709551616 — unquoted, two conversions.
Base and PR Python encoders: "18446744073709551616" — quoted, one conversion.

Retain the normalized integer and use it for both decimal formatting and the magnitude check. Add a regression test covering repeated conversion.

No other actionable findings. On Python 3.12.14, the full C/Python runner passed 462 tests, 70 skipped; another 2,320 targeted checks passed for boundaries, huge thresholds, nesting, keys, and ordinary subclasses. GitHub CI is awaiting workflow approval, so the remaining platform matrix is unverified.

**Review by etrepum — COMMENTED** [review]
py27 tests fail

```
    ======================================================================
    FAIL: test_large_bitcount_normalizes_subclass_once (simplejson.tests.test_bitsize_int_as_string.TestBitSizeIntAsString)
    ----------------------------------------------------------------------
    Traceback (most recent call last):
      File "/tmp/tmp.03GZKyJMIB/venv/lib/python2.7/site-packages/simplejson/tests/test_bitsize_int_as_string.py", line 153, in test_large_bitcount_normalizes_subclass_once
        value, int_as_string_bitcount=64, indent=indent))
    AssertionError: '"18446744073709551616"' != '1'
    
    ----------------------------------------------------------------------
    Ran 462 tests in 1.751s
    
    FAILED (failures=1, skipped=77)
    context canceled
```

File: simplejson/_speedups.c [hunk 1]
```diff
@@ -708,6 +710,30 @@ maybe_quote_bigint(PyEncoderObject* s, PyObject *encoded, PyObject *obj)
...
+        }
+        ge = PyObject_RichCompareBool(bits, s->large_int_bitcount, Py_GT);
+        Py_DECREF(bits);
+        if (ge < 0) {
+            Py_DECREF(encoded);
+            return NULL;
+        }
+        if (!ge)
+            return encoded;
+        goto quote;
```
Comment (etrepum) [inline simplejson/_speedups.c:734]: I think an else would be easier to follow than a goto/label pair, normally goto is only used for error handling

File: AGENTS.md [hunk 2]
```diff
@@ -240,3 +242,31 @@ per-module state in 4.0 was driven by free-threading and
 subinterpreter isolation, not by abi3 readiness. If you're adding a
 new feature, don't feel obligated to stay limited-API-compatible —
 performance-sensitive code wins.
+
+## Integer stringification thresholds
```
Comment (etrepum) [inline AGENTS.md:246]: I don't think we need to add text on this topic to AGENTS.md

File: CHANGES.txt [hunk 3]
```diff
@@ -1,3 +1,9 @@
+Unreleased
```
Comment (etrepum) [inline CHANGES.txt:1]: The changelog is updated manually before each release, you don't need to update this file

**Review by etrepum — APPROVED** [review]
